lvsfunc/comparison.py

```diff
@@ -169,42 +489,43 @@ def tvbd_diff(tv: vs.VideoNode, bd: vs.VideoNode,
     """
-    if thr > 128:
-        raise ValueError(f"tvbd_diff: \"thr\" should neither be nor exceed 128!'")
+    if thr >= 128:
+        raise ValueError("tvbd_diff: `thr` must be below 128")
+
+    if None in (tv.format, bd.format):
+        raise ValueError("tvbd_diff: variable-format clips not supported")
 
     tv, bd = depth(tv, 8), depth(bd, 8)
 
     if thr <= 1:
-        diff = core.std.PlaneStats(tv, bd)
-        frames = [i for i, f in enumerate(diff.frames()) if get_prop(f, "PlaneStatsDiff", float) > thr]
+        frames = [i for i, f in enumerate(core.std.PlaneStats(tv, bd).frames()) if get_prop(f, 'PlaneStatsDiff', float) > thr]
+        diff = core.std.MakeDiff(tv, bd)
     else:
         diff = core.std.MakeDiff(tv, bd).std.PlaneStats()
-        if diff.format is None:
-            raise ValueError("tvbd_diff: 'Variable-format clips not supported'")
-        t = float if diff.format.sample_type == vs.FLOAT else int
-        frames = [i for i, f in enumerate(diff.frames()) if get_prop(f, "PlaneStatsMin", t) <= thr or get_prop(f, "PlaneStatsMax", t) >= 255 - thr]
+        t = float if diff.format.sample_type == vs.SampleType.FLOAT else int
+        frames = [i for i, f in enumerate(diff.frames()) if get_prop(f, 'PlaneStatsMin', t) <= thr or get_prop(f, 'PlaneStatsMax', t) >= 255 - thr]
 
     if not frames:
-        raise ValueError(f"tvbd_diff: 'No differences found'")
+        raise ValueError("tvbd_diff: no differences found")
 
     if return_array:
-        return compare(tv.text.FrameNum().text.Text('Clip A', 9),
-                       bd.text.FrameNum().text.Text('Clip B', 9),
-                       frames)
+        tv, bd = tv.text.FrameNum(9), bd.text.FrameNum(9)
+        return Interleave(**{'TV': core.std.Splice([tv[f] for f in frames]), 'BD': core.std.Splice([bd[f] for f in frames])}).clip()
+
     else:
-        if thr <= 1:
-            diff = core.std.MakeDiff(tv, bd)
-        diff = core.resize.Spline36(diff, get_w(576), 576).text.FrameNum(8)
-        tv, bd = core.resize.Spline36(tv, int(diff.width / 2), int(diff.height / 2)), core.resize.Spline36(bd, int(diff.width / 2), int(diff.height / 2))
-        tv, bd = tv.text.Text("Clip A", 3), bd.text.Text("Clip B", 1)
-        stacked = core.std.StackVertical([core.std.StackHorizontal([tv, bd]), diff])
-        return core.std.Splice([stacked[f] for f in frames])
-
-
-# TODO: Write a comparison function that displays parts of clips side-by-side, similar to a slider.
-#       It should theoretically accept an infinite amount of clips
-#       and accurately split the width among all clips.
-#       Odd-resolution clips will also need to be taken into account.
+        diff = core.resize.Spline36(diff, get_w(height, only_even=False) * 2, height * 2).text.FrameNum(9)
+        tv, bd = tv.resize.Spline36(get_w(height, only_even=False), height).text.FrameNum(9), bd.resize.Spline36(get_w(height, only_even=False), height).text.FrameNum(9)
```

Comment:
```suggestion
        tv, bd = (c.resize.Spline36(get_w(height, only_even=False), height).text.FrameNum(9) for c in [tv, bd])
```

… or write it out in two lines so the code lines can be compared with the above `diff` easily.